tests/test_source.py

```diff
@@ -102,3 +106,24 @@ def test_filter():
     finally:
         import shutil
         shutil.rmtree(d)
+
+
+def test_with_cache():
+    import tempfile
+    d = tempfile.mkdtemp()
+    old = intake.config.conf['cache_dir']
+    expected = fastparquet.ParquetFile(os.path.join(here, 'split')).to_pandas()
+    try:
+        intake.config.conf['cache_dir'] = d
+        cat = intake.open_catalog(os.path.join(here, 'cache_cat.yaml'))
+        s = cat.split()
+        assert isinstance(s.cache[0], intake.source.cache.DirCache)
+        outfiles = s.cache[0].load(s._urlpath, output=False)
+        assert outfiles
+        assert outfiles[0].startswith(s.cache_dirs[0])
+        loc = s.cache[0]._path(s._urlpath)
+        assert glob.glob(loc + '/*/*/*.parquet')
+        assert s.read().equals(expected)
```

Comment:
You should clear the cache at the end of this test.

Reply:
I have the hard `rmtree` below - should be the same, but doesn't test the internals of the cache, right?